sonarts-core/src/rules/noUselessIntersectionRule.ts

```diff
@@ -47,11 +52,21 @@ export class Rule extends tslint.Rules.TypedRule {
 class Walker extends tslint.ProgramAwareRuleWalker {
   protected visitNode(node: ts.Node) {
     if (ts.isIntersectionTypeNode(node)) {
-      node.types.forEach(typeNode => {
-        if (isTypeWithoutMembers(this.getTypeChecker().getTypeFromTypeNode(typeNode))) {
-          this.addFailureAtNode(typeNode, Rule.MESSAGE);
-        }
-      });
+      const types = node.types.map(typeNode => ({
```

Comment:
@stas-vilchik I'm not sure how important it is for performance, but even for understanding IMO better move this `types` creation inside clause on line 64

Reply:
but I use it just on the next line!